### DISCUSSION TITLE
Develop DBeaver Plugin to create Azure SQL connection from database names listed in master database

### DISCUSSION THREAD

**Discussion by bnitelf** [dbeaver/dbeaver#19481] [DBeaver Development]
I want to develop my custom DBeaver plugin to create a new Azure SQL Connection from the database names listed in master database of Azure SQL Connection. 

So I want to ask you guys who have experience in this whether If it is possible. If so, then I will continue on research how to do it. Or if you have suggestion or reference repo, reference document on how to write plugin to communicate with DBeaver, I would be very appreciate. I have glanced through [Develop in Eclipse](https://github.com/dbeaver/dbeaver/wiki/Develop-in-Eclipse) and not sure if this tutorial is suited to my case.

I come from this issue about [Azure SQL driver does not support connect / view schemes / view tables in a single connection](https://github.com/dbeaver/dbeaver/issues/5258)

I'm working on Azure SQL database that have 30+ databases (and will increase every week). And it is nightmare for me to manually create every single of those databases. 

My workaround now is, I developed a powershell script that read excel file that contain connections info and then call dbeaver-cli.exe to create many connections better than create by DBeaver GUI.

**Comment by LonwoLonwo** [comment]
Hello @bnitelf 

Another way to resolve this issue with Azure is to look at the PostgreDatabase code and inheritance. DBeaver opens a new connection for any database in the data source under the hood.

**Reply by bnitelf** [reply]
OK Thank you for your reply, I will have a look at it.
